plugins/azure/fix_plugin_azure/resource/base.py

```diff
@@ -333,7 +344,55 @@ def collect_resources_in_group() -> None:
 
             self._resource_ids_in_group = [r["id"] for r in graph_builder.client.list(resources_api_spec)]
 
+        def collect_network_gateways() -> None:
+            from fix_plugin_azure.resource.network import AzureVirtualNetworkGateway
+
+            api_spec = AzureResourceSpec(
+                service="network",
+                version="2023-09-01",
+                path=f"{self.id}/providers/Microsoft.Network/virtualNetworkGateways",
+                path_parameters=[],
+                query_parameters=["api-version"],
+                access_path="value",
+                expect_array=True,
+            )
+            items = graph_builder.client.list(api_spec)
+            AzureVirtualNetworkGateway.collect(items, graph_builder)
+
+        def collect_local_network_gateway() -> None:
+            from fix_plugin_azure.resource.network import AzureLocalNetworkGateway
+
+            api_spec = AzureResourceSpec(
+                service="network",
+                version="2023-09-01",
+                path=f"{self.id}/providers/Microsoft.Network/localNetworkGateways",
+                path_parameters=[],
+                query_parameters=["api-version"],
+                access_path="value",
+                expect_array=True,
+            )
+            items = graph_builder.client.list(api_spec)
+            AzureLocalNetworkGateway.collect(items, graph_builder)
+
+        def collect_network_gateway_connections() -> None:
+            from fix_plugin_azure.resource.network import AzureVirtualNetworkGatewayConnection
+
+            api_spec = AzureResourceSpec(
+                service="network",
+                version="2023-09-01",
+                path=f"{self.id}/providers/Microsoft.Network/connections",
+                path_parameters=[],
+                query_parameters=["api-version"],
+                access_path="value",
+                expect_array=True,
+            )
+            items = graph_builder.client.list(api_spec)
+            AzureVirtualNetworkGatewayConnection.collect(items, graph_builder)
+
         graph_builder.submit_work(service_name, collect_resources_in_group)
+        graph_builder.submit_work(service_name, collect_network_gateways)
+        graph_builder.submit_work(service_name, collect_local_network_gateway)
+        graph_builder.submit_work(service_name, collect_network_gateway_connections)
```

Comment:
because they required resource group to fetch them:
https://learn.microsoft.com/en-us/rest/api/network-gateway/local-network-gateways/list?view=rest-network-gateway-2024-01-01&tabs=HTTP
https://learn.microsoft.com/en-us/rest/api/network-gateway/virtual-network-gateway-connections/list?view=rest-network-gateway-2024-01-01&tabs=HTTP
https://learn.microsoft.com/en-us/rest/api/network-gateway/virtual-network-gateways/list?view=rest-network-gateway-2024-01-01&tabs=HTTP